sympy/solvers/solveset.py

```diff
@@ -978,6 +989,354 @@ def _solveset(f, symbol, domain, _check=False):
+    be included in cases depending on their general form. Once figuring
+    out, different identifying helper conditions are included to check
+    which class the equation belongs. If the equation belongs to a
+    particular class it is solved by a call to its respective solving
+    helper. This helper either solves the equation completely or
+    transforms to a better form for `solveset` to handle. If the
+    equation is solved the result is returned otherwise a `ConditionSet`
+    is returned.
+
+
+    How `transolve` is better than `\_tsolve`
+    =========================================
+
+    1) Better output
+
+    `transolve` provides a better output to some equations than
+    `\_tsolve`. Though result from both the functions are correct
+    it's just that the one from `transolve` is easy to understand and
+    appropriate. This is because `transolve` smartly evaluates the
+    result minimising it wherever necessary.
+
+    Consider a simple exponential equation
+
+    >>> f = 3**(2*x) - 2**(x + 3)
+    >>> pprint(transolve(f, x, S.Reals), use_unicode=False)
+        -3*log(2)
+    {------------------}
+     -2*log(3) + log(2)
+    >>> pprint(tsolve(f, x), use_unicode=False)
+         /   3     \
+         | --------|
+         | log(2/9)|
+    [-log\2         /]
+
+
+    `transolve's` output seems more simple, readable and appropriate
+    as compared to that of `\_tsolve`.
+
+
+    2) Extensible
+
+    The API of `transolve` is designed such that it is easily extensible,
+    i.e, it is easy to add a new class of equation solver without having
+    to mess with the API or solving code of other helpers. The idea
+    behind transolve is making seperate helpers for different equation
+    solver. So to make new class of equation solve, you must add a
+    helper that identifies that particular class of the equation and
+    another helper that solves that type of equations. This way it
+    becomes easily extensible.
+
+
+    3) Modular
+
+    `transolve` is designed to be modular i.e, for every class of
+    equation a separate helper for identification and solving is
+    implemented. This makes it easy to change or modify any of the
+    method implemented directly in the helpers without interfering
+    with the actual structure of the API.
+
+
+    4) Less complex API
+
+    Since modular approach is used in transolve, the API automatically
+    becomes less complex to read and understand unlike in `\_tsolve`
+    where the API structure is a mess and lots of recursive and function
+    calls makes it more hard to understand.
+
+
+    5) Faster Computation
+
+    Equation solving via transolve is much faster as compared to
+    `\_tsolve`. The way equations are treated in solve is complicated
+    which make it a bit time consuming. `solve` tries every possibility
+    to solve the equation and in process it gets into each and every
+    condition and starts solving the equation to get the result,
```

Comment:
`The way equations are treated to solve is complicated` -> This doesn't really provide any useful information.


You should rephrase the following:
```
`solve` tries every possibility
+    to solve the equation and in process it gets into each and every
+    condition and starts solving the equation to get the result,
```